### PR TITLE
Save asset in Firebase

### PR DESCRIPTION
**Pull request by ignaciocervinoGlob**
### Description
Saving asset in Firebase:
- FormViewModel calls SaveAssetUseCaseProtocol
- SaveAssetUseCaseProtocol execute the use case and calls the saveAsset in AssetProviderProtocol
- AssetProviderProtocol executes saveAsset service that makes the save in Firebase, it returns success if the asset is saved correctly or failure if something wrong happens

#### Tasks:
- [x] Add internet validation
- [x] In Add asset mode in the form, if all validations are passed, save the information in db
- [x] Create SaveAssetUseCase and its protocol
- [x] Create the service protocol and its implementation
- [x] Unit tests

#### No internet case 
https://user-images.githubusercontent.com/90403672/152033135-6ea12aa7-4c15-44a4-bf0c-0f15ecbdce59.mp4

Disclaimer: To test this one I disconnected the Mac from the internet since we can do it in the Xcode emulator

#### Success case
https://user-images.githubusercontent.com/90403672/152033248-4d6cfc9a-c7f4-449a-bdd9-8db2028622cc.mp4

image

### Coverage
image
image

### CODE REVIEW

File: stockApp/stockApp/Presentation/ViewModel/FormViewModel.swift [hunk 1]
```diff
@@ -45,17 +45,20 @@ final class FormViewModel: FormViewModelProtocol {
...
-                                 location: globerMail,
-                                 userMail: location
-        ) { result in
-            DispatchQueue.main.async {
-                switch result {
-                    case .success:
-                        self.goToPreviousWindow?()
-                    case .failure:
-                        self.showErrorWindow?(Strings.errorFirebase)
+        DispatchQueue.global(qos: .background).async { [self] in
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/ViewModel/FormViewModel.swift:48]: use weak self instead of self

**Review by ezequielvelezglobant — CHANGES_REQUESTED** [review]

File: stockApp/stockApp/Presentation/ViewModel/FormViewModel.swift (hunk 1 above)
Comment (ignaciocervinoGlob) [reply stockApp/stockApp/Presentation/ViewModel/FormViewModel.swift]: done

File: stockApp/stockApp/Presentation/Utility/HelperFunctions.swift [hunk 2]
```diff
@@ -2,44 +2,56 @@ import Foundation
...
-				viewController.dismiss(animated: false, completion: nil)
-			}))
-			alertMessage.view.tintColor = Color.tundoraBlack
-			viewController.present(alertMessage, animated: true, completion: nil)
-		}
-	}
+            DispatchQueue.main.async {
+                let alertMessage = UIAlertController(title: "\(title)", message: "\(message)", preferredStyle: .alert)
+                alertMessage.addAction(UIAlertAction(title: Strings.searchButtonMessage,
+                                                     style: .default, handler: {_ in
```
[the remark this answers is not in the document]
Comment (ignaciocervinoGlob) [reply stockApp/stockApp/Presentation/Utility/HelperFunctions.swift]: done

File: stockApp/stockApp/Presentation/Utility/HelperFunctions.swift [hunk 3]
```diff
@@ -2,44 +2,56 @@ import Foundation
...
+        title: String = Strings.emptyString,
+        message: String = Strings.emptyString,
+        viewController: UIViewController,
+        barcode: String,
+        assetName: String,
+        globerMail: String,
+        location: String) {
+            let alertMessage = UIAlertController(title: title, message: message, preferredStyle: .alert)
+            alertMessage.addAction(UIAlertAction(title: Strings.cancelMessage, style: .default, handler: nil))
+            alertMessage.addAction(UIAlertAction(title: Strings.okMessage, style: .default, handler: {_ in
```
[the remark this answers is not in the document]
Comment (ignaciocervinoGlob) [reply stockApp/stockApp/Presentation/Utility/HelperFunctions.swift]: done

**Review by AgustinaB-Globant — APPROVED** [review]

**Review by ezequielsetzesglobant — APPROVED** [review]

**Review by ezequielvelezglobant — APPROVED** [review]

**Pull request merged**